src/main/java/Duke.java

```diff
@@ -1,10 +1,129 @@
+import java.util.Scanner;
 public class Duke {
+
+    public static Task[] addTask(Task[] tasks, Task task){
+        tasks = Arrays.copyOf(tasks, tasks.length + 1);
+        tasks[tasks.length - 1] = task;
+        return tasks;
+    }
+
+    public static String joinString(String[] strings, int idx){
+        String joinedString = "";
+        for (int i = idx; i < strings.length; i++){
+            if (strings[i] == null){
+                break;
+            } else {
+                joinedString += strings[i] + " ";
+            }
+        }
+        return joinedString.trim();
+    }
+    public static void welcome(){
+        System.out.println("Hello! I'm Duke\nWhat can I do for you?");
+    }
+    public static void bye(){
+        System.out.println("Adios!");
+    }
+    public static void list(Task[] arr){
+        System.out.println("Here are the tasks in your list:");
+        int ID = 1;
+        for (Task tsk : arr) {
+            if (tsk==null){break;}
+            System.out.println(ID + "." + printIconStatus(tsk));
+            ID += 1;
+        }
+    }
+
+    public static void unmark(String[] strings, Task[] tasks){
+        if (Integer.parseInt(strings[1]) > tasks.length){
+            System.out.println("Item not found!");
+        } else {
+            tasks[Integer.parseInt(strings[1]) - 1].markAsNotDone();
+            System.out.println("OK, I've marked this task as not done yet:");
+            Task tsk = tasks[Integer.parseInt(strings[1]) - 1];
+            System.out.println(printIconStatus(tsk));
+        }
+    }
+
+    public static void mark(String[] strings, Task[] tasks){
+        if (Integer.parseInt(strings[1]) > tasks.length){
+            System.out.println("Item not found!");
+        } else {
+            tasks[Integer.parseInt(strings[1]) - 1].markAsDone();
+            System.out.println("Nice! I've marked this task as done:");
+            Task tsk = tasks[Integer.parseInt(strings[1]) - 1];
+            System.out.println(printIconStatus(tsk));
+        }
+    }
+
+    public static String printIconStatus(Task tsk){
+        // if (tsk instanceof [Class]) to check if tsk is of a certain class
+        String out = "[" + tsk.getIcon() + "][" + tsk.getStatusIcon() + "] " + tsk.description;
+        if (tsk instanceof Events){
+            out += " (" + tsk.getStart() + " " + tsk.getEnd() + ")";
+        } else if (tsk instanceof  Deadlines) {
+            out += " (" + tsk.getDeadline() + ")";
+        }
+        return out;
+    }
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        welcome();
+        String input;
+        Task[] tasks = new Task[0];
+        int lstID = 1;
+        while (true) {
```

Comment:
I notice that you have a tendency to use magic strings and numbers. Do try to avoid that where possible, and instead use constants to set your strings/numbers.